### DISCUSSION TITLE
Netbox 3.1 - Server Error - Azure AD

### DISCUSSION THREAD

**Discussion by RickEXV** [netbox-community/netbox#9165] [Q&A] [answered]
Hi,

We have followed this guide for installation: <https://pypi.org/project/netbox-plugin-azuread/>
Sadly we end up with this error : 

```
<class 'AttributeError'>

'NoneType' object has no attribute 'get'

Python version: 3.8.10
NetBox version: 3.1.8

```

We have the Azure AD login button on our site and we get redirected properly for authentication. When we fill in our credentials we get redirected to our Netbox instance with the following page:

![image](https://user-images.githubusercontent.com/103999806/163984115-cc75f340-f3f3-4a23-8acf-5ede50c0c292.png)

Help is much appreviated!

**Comment by jeremystretch** [comment]
You don't need a separate plugin for Azure AD to work; NetBox supports this natively. You can follow [this guide](https://netbox.readthedocs.io/en/develop/administration/authentication/microsoft-azure-ad/) to configure it.

**Reply by RickEXV** [reply]
> You don't need a separate plugin for Azure AD to work; NetBox supports this natively. You can follow [this guide](https://netbox.readthedocs.io/en/develop/administration/authentication/microsoft-azure-ad/) to configure it.

Hi Jeremy,

Thank you for your help.

I followed the guide and i now receive a new error. Which is the following:

AADSTS50011: The redirect URI 'http://<domain>/oauth/complete/azuread-oauth2/' specified in the request does not match the redirect URIs configured for the application '3da21e22-a3a2-482f-8476-e99fd3285f5c'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. Navigate to https://aka.ms/redirectUriMismatchError to learn more about how to fix this.

We have the same URL defined in the Azure app registration:

![image](https://user-images.githubusercontent.com/103999806/164007571-2d2f776e-af76-4247-aec6-a85fad1e1d12.png)


Hope to hear from you!

**Reply by jeremystretch** [reply]
Per the Azure instructions, the URL must start with `https://` if not referencing localhost.

**Comment by RickEXV** [comment, marked as answer]
We have followed the guide and the only place where to fill in an URL is at the Redirect URLs form. In the screenshot you can also see we provide HTTPS. Is it possible we have to change it somewhere else?

**Reply by jeremystretch** [reply]
You need to be running NetBox via HTTPS. If you have a load balancer or something in front of NetBox that's connecting via HTTP, you can set `SOCIAL_AUTH_REDIRECT_IS_HTTPS = True` in `configuration.py` to force the authentication library to use HTTPS when sending the redirect URI (see the [python-social-auth docs](https://python-social-auth.readthedocs.io/en/latest/configuration/settings.html#processing-redirects-and-urlopen)).

**Reply by RickEXV** [reply]
I added the SOCIAL_AUTH_REDIRECT_IS_HTTPS = True line and it all works fine now!

Thanks alot!

**Reply by jeremystretch** [reply]
I just added a note to the guide mentioning this.
